woocommerce/woocommerce-ios
# [SwiftUI Payments Menu] Show selected payment gateway

Closes: #11344 


## Description


Previously we forced a choice in the Payment Provider screen, without showing what they already had selected.

This wasn’t very clear. Furthermore we didn’t update the row on the payments menu to reflect their new choice until the onboarding was successful for that gateway, which made it look like we hadn’t made the change.

This PR updates the Payment Provider screen to show the currently selected plugin, where there is one, and updates the row according to the selected plugin even when onboarding is incomplete.

## Testing instructions


1. Launch the app, and switch to a store with both Woo Payments and Stripe Gateway activated.
2. Navigate to `Menu > Payments`
3. Tap the Payment Provider row
4. Observe that your selected gateway is marked in the screen which shows
5. Change your selection
6. Observe that (after a moment) the Payment Provider row is updated with your selection; even if onboarding is not completed for the second plugin.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/2472348/c1253e6f-4858-44ce-87c5-58a3c02b2c84



---
